units/crons.go

```diff
@@ -846,19 +851,25 @@ func PopulateHostCreationJobs(env evergreen.Environment, part int) amboy.QueueOp
 			if h.UserHost || h.SpawnOptions.SpawnedByTask {
 				// pass:
 				//    always start spawn hosts asap
+
 			} else {
-				if submitted > env.Settings().HostInit.HostThrottle {
-					// throttle hosts, so that we're starting very
-					// few hosts on every pass. Hostinit runs very
-					// frequently, lets not start too many all at
-					// once.
-					continue
+				num, _ := distroCount[h.Distro.Id]
+				if num == 0 || num < len(runningHosts)/100 {
```

Comment:
but maybe this will give priority to the distros for which there's the least demand?

Reply:
I think that's inevitable. My hope is that this is conservative enough to have a reasonably small impact on high-demand distros.